scm/etc/scripts/UFS_forcing_ensemble_generator.py

```diff
@@ -0,0 +1,193 @@
+        npts = args.nensmembers
+        if (args.lat_list or args.lon_list):
+            print("ERROR: Can't provide explicit lon/lat range AND number of points for ensemble generation.")
+            exit()
+    else:
+        if (args.lon_list and args.lat_list):
+            if (len(args.lon_list) == len(args.lat_list)):
+                npts = len(args.lon_list)
+            else:
+                print("ERROR: Number of longitude/latitudes are inconsistent")
+                exit()
+
+    ###########################################################################
+    #
+    # Set longitude/latitude 
+    #
+    ###########################################################################
+    if (args.nensmembers):
+        rng1 = np.zeros((npts), dtype=float)
+        rng2 = np.zeros((npts), dtype=float)
+        lons = np.zeros((npts), dtype=float)
+        lats = np.zeros((npts), dtype=float)
+        for ipt in range(npts):
+            # Here the seed is set to give the same set of points each time.
+            random.seed(ipt)
+            rng1[ipt] = random.randint(1,1000)*0.001
+            rng2[ipt] = random.randint(2,1000)*0.001
+            #
+            if args.lat_limits:
+                lats[ipt] = args.lat_limits[0] + (args.lat_limits[1]-args.lat_limits[0])*rng1[ipt]
+            else:
+                lats[ipt] = rng1[ipt]*180-90
+            if args.lon_limits:
+                lons[ipt] = args.lon_limits[0] + (args.lon_limits[1]-args.lon_limits[0])*rng2[ipt]
+            else:
+                lons[ipt] = rng2[ipt]*360
+    ###########################################################################
+    #
+    # Use longitude and latitude provided
+    #
+    ###########################################################################
+    else:
+        lons = np.asarray(args.lon_list)
+        lats = np.asarray(args.lat_list)
+
+    ###########################################################################
+    #
+    # Create SCM case configuration (etc/case_config) file.
+    #
+    ###########################################################################
+    #
+    case_config =[{"name": "input_type",  "values": str(1)},             \
+                  {"name": "dt",          "values": str(args.timestep)}, \
+                  {"name": "C_RES",       "values": str(args.C_RES)}]
+
+    # What, if any, options neeed to be passsed to UFS_IC_generator.py?
+    com_config = ''
+    if args.save_comp:   com_config = com_config + ' -sc'
+    if args.use_nearest: com_config = com_config + ' -near'
+
+    # Create inputs to SCM
+    case_list    = ""
+    case_list_nf = ""
+    count = 0
+    run_list = []
+    for pt in range(0,npts):
+        # Call UFS_IC_generator.py
+        case_name     = args.case_name +"_n" + str(pt).zfill(3)
+        file_scminput = "../../data/processed_case_input/"+case_name+"_SCM_driver.nc"
+        com = "./UFS_IC_generator.py -l " +str(lons[pt]) + " " + str(lats[pt]) + \
```

Comment:
Is this slow? Would it benefit from using multiprocessing, e.g. a solution like the `execute` function in the run script?

Reply:
This refers to using the `os.system(com)` serially, below.